elements/dash-tabs/dash-tabs.js

```diff
@@ -0,0 +1,293 @@
+      // and get the modulus to wrap around if necessary.
+      return tabs[(newIdx + tabs.length) % tabs.length];
+    }
+
+    /**
+     * `_nextTab` gets the tab that comes after the currently selected one,
+     * wrapping around when reaching the last tab.
+     */
+    _nextTab() {
+      const tabs = this._allTabs();
+      let newIdx =
+        tabs.findIndex(tab =>
+          tab.getAttribute('aria-selected') === 'true') + 1;
+      return tabs[newIdx % tabs.length];
+    }
+
+    /**
+     * `reset` marks all tabs as deselected and hides all the panels.
+     */
+    reset() {
+      const tabs = this._allTabs();
+      const panels = this._allPanels();
+
+      tabs.forEach(tab => {
+        tab.tabIndex = -1;
+        tab.setAttribute('aria-selected', 'false');
+      });
+
+      panels.forEach(panel => {
+        panel.setAttribute('aria-hidden', 'true');
+      });
+    }
+
+    /**
+     * `disconnectedCallback` removes the event listeners that
+     * `connectedCallback` added.
+     */
+    disconnectedCallback() {
+      this.removeEventListener('keydown', this._onKeyDown);
+      this.removeEventListener('click', this._onClick);
+    }
+
+    /**
+     * `_selectTab` marks the given tab as selected.
+     * Additionally, it unhides the panel corresponding to the given tab.
+     */
+    _selectTab(newTab) {
+      // Deselect all tabs and hide all panels.
+      this.reset();
+
+      // Get the panel that the `newTab` is associated with.
+      const newPanel = this._panelForTab(newTab);
+      // If that panel doesn’t exist, abort.
+      if (!newPanel) throw new Error(`No panel with id ${newPanelId}`);
+
+      // Unhide the panel and mark the tab as active.
+      newPanel.setAttribute('aria-hidden', 'false');
+      newTab.setAttribute('aria-selected', 'true');
+      newTab.tabIndex = 0;
+      newTab.focus();
+    }
+
+    /**
+     * `_onKeyDown` handles key presses inside the tab panel.
+     */
+    _onKeyDown(event) {
+      // If the keypress did not originate from a tab element itself,
+      // it was a keypress inside the a panel or on empty space. Nothing to do.
+      if (event.target.getAttribute('role') !== 'tab') return;
+      // Don’t handle modifier shortcuts typically used by assistive technology.
+      if (event.altKey) return;
+
+      // The switch-case will determine which tab should be marked as active
+      // depending on the key that was pressed.
+      let newTab;
+      switch (event.keyCode) {
+        // When up or left is pressed, the previous tab is selected.
+        case KEYCODE.LEFT:
+        case KEYCODE.UP:
+          newTab = this._prevTab();
+          break;
+
+        // When right or down is pressed, it’s the next tab.
+        case KEYCODE.RIGHT:
+        case KEYCODE.DOWN:
+          newTab = this._nextTab();
+          break;
+
+        // The home button always selects the first tab.
+        case KEYCODE.HOME:
+          newTab = this._allTabs()[0];
```

Comment:
OK, we can switch to function calls then.

So... what do you think about a _firstTab() and _lastTab() ?  😸